container_api/src/container.rs

```diff
@@ -119,19 +163,58 @@ impl Container {
         f.read_to_string(&mut contents)?;
         Dna::try_from(JsonString::from(contents))
     }
+
+    fn make_dispatcher(&self, interface_config: &InterfaceConfiguration) -> RpcDispatcher {
+        let InterfaceConfiguration {
+            id,
+            driver,
+            admin,
+            instances,
+        } = interface_config;
+        let instance_ids: Vec<String> = instances.iter().map(|i| i.id.clone()).collect();
+        let instance_subset: InstanceMap = self
+            .instances
+            .iter()
+            .filter(|(id, _)| instance_ids.contains(&id))
+            .map(|(id, val)| (id.clone(), val.clone()))
+            .collect();
+        RpcDispatcher::new(instance_subset)
+    }
+
+    fn spawn_interface_thread(
+        &self,
+        interface_config: InterfaceConfiguration,
+    ) -> InterfaceThreadHandle {
+        let dispatcher = self.make_dispatcher(&interface_config);
+        thread::spawn(move || {
+            let iface = make_interface(&interface_config);
+            iface.run(Arc::new(dispatcher))
+        })
+    }
 }
 
 impl<'a> TryFrom<&'a Configuration> for Container {
     type Error = HolochainError;
     fn try_from(config: &'a Configuration) -> Result<Self, Self::Error> {
-        let mut container = Container::new();
+        let mut container = Container::with_config((*config).clone());
```

Comment:
yes this is redundant, need to clean up the API more to prevent loading the config twice